tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -311,341 +329,342 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
-				return false, nil
-			}
-			return true, nil
-		})
-		framework.ExpectNoError(err, "failed to wait for TidbCluster managed svc to be ready: %q", tc.Name)
-	})
+				owner := metav1.GetControllerOf(svc)
+				if owner == nil {
+					log.Logf("tidb service has not been adopted by TidbCluster yet")
+					return false, nil
+				}
+				framework.ExpectEqual(metav1.IsControlledBy(svc, tc), true, "Expected tidb service owner is TidbCluster")
+				framework.ExpectEqual(svc.Spec.ClusterIP, oldSvc.Spec.ClusterIP, "tidb service ClusterIP should be stable across adopting and updating")
+				return true, nil
+			})
+			framework.ExpectNoError(err, "failed to wait for TidbCluster managed svc to be ready: %q", tc.Name)
 
-	// Basic IT for managed in TidbCluster CR
-	// TODO: deploy pump through CR in backup and restore
-	// TODO: Add pump configmap rolling-update case
-	ginkgo.It("should adopt helm created pump with TidbCluster CR", func() {
-		ginkgo.By("Deploy initial tc")
-		tcCfg := newTidbClusterConfig(e2econfig.TestConfig, ns, "pump", "admin", utilimage.TiDBV4)
-		tcCfg.Resources["pd.replicas"] = "1"
-		tcCfg.Resources["tikv.replicas"] = "1"
-		tcCfg.Resources["tidb.replicas"] = "1"
-		oa.DeployTidbClusterOrDie(&tcCfg)
-		oa.CheckTidbClusterStatusOrDie(&tcCfg)
+			ginkgo.By("Updating TiDB service")
+			trafficPolicy := corev1.ServiceExternalTrafficPolicyTypeLocal
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Spec.TiDB.Service.Type = corev1.ServiceTypeNodePort
+				tc.Spec.TiDB.Service.ExternalTrafficPolicy = &trafficPolicy
+				tc.Spec.TiDB.Service.Annotations = map[string]string{
+					"test": "test",
+				}
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update tidb service of TidbCluster: %q", tc.Name)
 
-		ginkgo.By("Deploy pump using helm")
-		err := oa.DeployAndCheckPump(&tcCfg)
-		framework.ExpectNoError(err, "failed to deploy pump for TidbCluster: %v", tcCfg)
+			ginkgo.By("Waiting for the TiDB service to be synced")
+			err = wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+				svc, err := c.CoreV1().Services(ns).Get(controller.TiDBMemberName(tcName), metav1.GetOptions{})
+				if err != nil {
+					if errors.IsNotFound(err) {
+						return false, err
+					}
+					log.Logf("error get TiDB service: %v", err)
+					return false, nil
+				}
+				if isEqual, err := gomega.Equal(corev1.ServiceTypeNodePort).Match(svc.Spec.Type); !isEqual {
+					log.Logf("tidb service type is not %s, %v", corev1.ServiceTypeNodePort, err)
+					return false, nil
+				}
+				if isEqual, err := gomega.Equal(trafficPolicy).Match(svc.Spec.ExternalTrafficPolicy); !isEqual {
+					log.Logf("tidb service traffic policy is not %s, %v", svc.Spec.ExternalTrafficPolicy, err)
+					return false, nil
+				}
+				if haveKV, err := gomega.HaveKeyWithValue("test", "test").Match(svc.Annotations); !haveKV {
+					log.Logf("tidb service has no annotation test=test, %v", err)
+					return false, nil
+				}
```

Comment:
suggest change to plain golang comparison for better readability, like `if corev1.ServiceTypeNodePort != svc.Spec.Type`